lib/cosmos/packets/structure.rb

```diff
@@ -327,7 +332,37 @@ def buffer=(buffer)
     # Take the structure mutex to ensure the buffer does not change while you perform activities
     def synchronize
       @mutex ||= Mutex.new
-      @mutex.synchronize {|| yield }
+      @mutex.synchronize {|| yield}
+    end
+
+    # Take the structure mutex to ensure the buffer does not change while you perform activities
+    # This versions allows reads to happen if a top level function has already taken the mutex
```

Comment:
By "top level" do you mean only methods within Structure and Packet? If so we should say that. This is clever and I see why you had to do it but I worry it is confusing and that over time we will miss places it should be used or why it was used.